### ISSUE TITLE
Bug - List variables are not converted to the right types.

### ISSUE THREAD

**Issue by ddelegator** [aPureBase/KGraphQL#138]
Query:
```graphql
mutation ChannelMutations($somethings: [SomethingInput!]!) {
    createSomethings(somethings: $somethings) {
        result
    }
}
```
variables:
```json
{ "something": [{"name": "a", "value": 10}, {"name": "b", "value": 20}, {"name": "c", "value": 30}] }
```

What I expect:
I expect this to generate a list of SomethingInput

What happens:
This generates a list of maps.

And this is the error I get later in my code because I'm trying to use the map as `SomethingInput`:
```
java.lang.ClassCastException: java.util.LinkedHashMap cannot be cast to SomethingInput
```

The query without variables works as expected:
```graphql
mutation ChannelMutations {
    createSomethings(
        somethings: [{name: "a", value: 10}, {name: "b", value: 20}]
    ) {
        result
    }
}
```

**Comment by jeggy** [comment]
What version of KGraphQL are you using? 
This is the exact issue I tried to fix with version `0.17.4`.

**Comment by ddelegator** [comment]
I'm using  16.something so upgrading is probably the solution. Thank you! I'm going to close this issue.
